[5.1] Enable CallbackEvent to save output to file

ikari7789 opened the pull request:
This will allow closure based Scheduler events to save their output to file if the closure returns a string.

GrahamCampbell commented on src/Illuminate/Console/Scheduling/CallbackEvent.php:
```diff
@@ -67,6 +68,24 @@ public function run(Container $container)
+        }
+
+        $openMode = $this->shouldAppendOutput ? 'a' : 'w';
+        $writeFile = new \SplFileObject($this->output, $openMode);
```
please import this

ikari7789 replied:
Done.

GrahamCampbell commented on src/Illuminate/Console/Scheduling/CallbackEvent.php:
```diff
@@ -4,6 +4,7 @@
 
 use LogicException;
 use InvalidArgumentException;
+use SplFileObject;
```
Please sort all imports by length.

ikari7789 replied:
Ah, sorry. Did it alphabetically/by depth. Done.

GrahamCampbell commented on src/Illuminate/Console/Scheduling/CallbackEvent.php:
```diff
@@ -67,6 +69,24 @@ public function run(Container $container)
     }
 
     /**
+     * Write the response to file.
```
to a file

GrahamCampbell replied:
or "to the disk" or something

ikari7789 replied:
"Write the given response to the disk."
Done?

GrahamCampbell replied:
:)

GrahamCampbell commented on src/Illuminate/Console/Scheduling/CallbackEvent.php:
```diff
@@ -67,6 +69,24 @@ public function run(Container $container)
+        $openMode = $this->shouldAppendOutput ? 'a' : 'w';
+        $writeFile = new SplFileObject($this->output, $openMode);
+        $writeFile->fwrite($response);
+        $writeFile = null;
```
not needed

GrahamCampbell replied:
php's gc will deal with this

ikari7789 replied:
Done.

GrahamCampbell commented on src/Illuminate/Console/Scheduling/CallbackEvent.php:
```diff
@@ -67,6 +69,23 @@ public function run(Container $container)
+
+        $openMode = $this->shouldAppendOutput ? 'a' : 'w';
+        $writeFile = new SplFileObject($this->output, $openMode);
+        $writeFile->fwrite($response);
```
Writing `(new SplFileObject($this->output, $openMode))->fwrite($response)` would prevent the object hanging around I think.

ikari7789 replied:
Works for me. Done.

ikari7789 committed:
Enable CallbackEvent to save output to file
Allows closure based Scheduler events to save their output to file if the closure returns a string.

GrahamCampbell commented on src/Illuminate/Console/Scheduling/CallbackEvent.php:
```diff
@@ -67,6 +69,22 @@ public function run(Container $container)
+        }
+
+        $openMode = $this->shouldAppendOutput ? 'a' : 'w';
+        (new SplFileObject($this->output, $openMode))->fwrite($response);
```
maybe add a blank line before this for readability

GrahamCampbell replied:
sorry for messing you about

taylorotwell commented:
If you need to save stuff to a file use the File methods from the Closure.

ikari7789 commented:
I don't see how using the File methods from the Closure eliminates the need for this pull request. All file/mail operations related to a scheduler event currently only work with ```exec()``` and ```command()```. It does not work at all if you use ```call()```. It hinders a rather useful feature and works against how the person developing an application would expect it to work.
